src/Management/test/Endpoint.Test/CloudFoundry/SecurityBaseTest.cs

```diff
@@ -49,7 +50,7 @@ private static SecurityBase GetSecurityBase(out CloudFoundryEndpointOptions clou
     {
         cloudOpts = GetOptionsFromSettings<CloudFoundryEndpointOptions>();
         managementOptions = GetOptionsMonitorFromSettings<ManagementEndpointOptions>().Get(CFContext.Name);
-        var securityBase = new SecurityBase(cloudOpts, managementOptions);
+        var securityBase = new SecurityBase(cloudOpts, managementOptions, NullLogger.Instance);
```

Comment:
Unrelated to this PR, and perhaps a silly question: Looking at the name, `SecurityBase` seems to be intended as a base class. However, it has no virtual and no protected members. And there are no types in the solution that inherit from it. Which makes me wonder: what is the intended purpose of this class?

Reply:
Good catch, it used to be inheritance but got changed and it doesn't need to be public.

Reply:
In that case, can we make the type `internal sealed` with a public constructor? There's already an `[InternalsVisibleTo]` so tests can still access it.

What I learned the hard way, is that it would have saved me a lot of work if I'd started out with making types internal and/or sealed before looking at the Sonar violations. Many of them only apply to the public API surface, and some don't apply to sealed types. For each type, the basic question is: do we intend Steeltoe users to inherit from it? If so, it should have protected and/or virtual members. By narrowing down our public API, we're getting a lot more freedom to change things without having to wait for the next major version.